Tasks/Gradle/CodeAnalysis/pmd.gradle

```diff
@@ -0,0 +1,31 @@
+rootProject {
+    afterEvaluate {project ->
+        def hasPmd = false;
+
+        allprojects.forEach{ p ->
+
+            if (p.plugins.hasPlugin('pmd')) {
+                hasPmd = true;
+                println('$p is already configured to run PMD. Skipping auto configuration ');
+            }
+        }
+
+        if (!hasPmd) {
+            println('Applying the PMD plugin to all projects');
+            allprojects.forEach{p->
+                p.apply plugin: 'pmd'
+            }
+        }
+
+        allprojects { p->
+            if (!hasPmd) {
+                pmd {
+                    ignoreFailures = true;
```

Comment:
By default  PMD violations break the build so I need to instruct the plugin to just create the reports.